Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,371 @@ import UIKit
+
         return headerView
     }
-    
+
+    @objc func inputOutputSectionHeaderViewTapped(_ sender: UITapGestureRecognizer?) {
+        guard let section = sender?.view?.tag else { return }
+
+        if section == 1 {
+            self.isTxInputsCollapsed = !self.isTxInputsCollapsed
+            self.transactionDetailsTable.reloadData()
+        } else if section == 2 {
+            self.isTxOutputsCollapsed = !self.isTxOutputsCollapsed
+            self.transactionDetailsTable.reloadData()
+        }
+    }
+
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         switch indexPath.section {
         case 0:
             let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionDetailCell") as! TransactionDetailCell
-            cell.txnDetails = self.generalTxDetails[indexPath.row]
+            cell.txDetail = self.generalTxDetails[indexPath.row]
+            cell.onTxDetailValueCopied = { copiedDetail in
+                Utils.showBanner(parentVC: self, type: .success, text: String(format: LocalizedStrings.sgCopied, copiedDetail))
+            }
             return cell
-            
+
         case 1:
-            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactiontInputDetailsCell") as! TransactiontInputDetailsCell
-            cell.setup(transaction.inputs, presentingController: self)
-            cell.expandOrCollapse = { [weak self] in
-                self?.tableTransactionDetails.reloadData()
-            }
+            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionInputDetailsCell") as! TransactionInputDetailCell
+            cell.setup(transaction.inputs[indexPath.row])
             return cell
-            
+
         case 2:
-            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactiontOutputDetailsCell") as! TransactiontOutputDetailsCell
-            cell.setup(transaction.outputs, presentingController: self)
-            cell.expandOrCollapse = { [weak self] in
-                self?.tableTransactionDetails.reloadData()
-            }
+            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionOutputDetailsCell") as! TransactionOutputDetailCell
+            cell.setup(transaction.outputs[indexPath.row])
             return cell
-            
+
+        case 3:
+            return tableView.dequeueReusableCell(withIdentifier: "TransactionViewOnDcrdataCell")!
+
         default:
             return UITableViewCell()
         }
     }
-    
+
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if indexPath.section == 0 && indexPath.row == 6 {
-            self.copyText(self.transaction.hash)
+        if indexPath.section == 1, let cell = tableView.cellForRow(at: indexPath) as? TransactionInputDetailCell {
+            DispatchQueue.main.async {
+                UIPasteboard.general.string = cell.txHashLabel.text
```

Comment:
I don't think this copy action should be on tap of the entire cell. Instead, set up a tap gesture recognizer on the `txHashLabel` in the `TransactionInputDetailCell` or `TransactionOutputDetailCell`.